pygments/lexers/ocl.py

```diff
@@ -0,0 +1,207 @@
+        'String', 
+        'OrderedSet', 
+        'Tuple',
+        'Bag',
+        'Set',
+        'Sequence',
+        'OclInvalid', 
+        'OclVoid',
+        'TupleType', 
+        'OclState',
+        'Collection',
+        'OclMessage'
+    ]
+
+    _functions = [
+        # OclAny Operations
+        'oclAsSet',
+        'oclIsNew',
+        'oclIsUndefined',
+        'oclIsInvalid',
+        'oclAsType',
+        'oclIsTypeOf',
+        'oclIsKindOf',
+        'oclInState',
+        'oclType',
+        'oclLocale',
+        # OclMessage Operations
+        'hasReturned',
+        'result',
+        'isSignalSent',
+        'isOperationCall'
+        # Real Operations
+        'abs',
+        'floor',
+        'round',
+        'max',
+        'min',
+        'toString',
+        # Integer Operations
+        'div',
+        'mod',
+        # String Operations
+        'size',
+        'substring',
+        'concat',
+        'toInteger',
+        'toReal',
+        'toUpperCase',
+        'toLowerCase',
+        'indexOf',
+        'equalsIgnoreCase',
+        'at',
+        'characters',
+        'toBoolean',
+        # Collection Operations
+        'includes',
+        'excludes',
+        'count',
+        'includesAll',
+        'excludesAll',
+        'isEmpty',
+        'notEmpty',
+        'sum',
+        'product',
+        'selectByKind',
+        'selectByType',
+        'asBag',
+        'asSequence',
+        'asOrderedSet',
+        'asSet',        
+        'flatten',
+        # Set Operations
+        'union',
+        'intersection',
+        'including',
+        'excluding',
+        'symmetricDifference'
+        'count',
+        # OrderedSet Operations
+        'append',
+        'prepend',
+        'insertAt',
+        'subOrderedSet',
+        'first',
+        'last',
+        'reverse',
+        # Bag Operations
+        # Sequence Operations
+        'subSequence',
+        # Iterator Expressions
+        'any',
+        'closure',
+        'collect',
+        'collectNested',
+        'exists',
+        'forAll',
+        'isUnique',
+        'iterate',
+        'one',
+        'reject',
+        'select',
+        'sortedBy',
+        # Predefined operations
+        'allInstances',
+        # Controversial
+        'average',
+        'conformsTo'
+    ]
+
+    _operator = [
+        'or', 
+        'xor',
+        'and',
+        'not',
+        'implies'
+    ]
+
+    tokens = {
+        'root': [
+            (r'\n', Text),
+            (r'[^\S\n]+', Text),
+            (r'--.*?\n', Comment.Single),
+            (r'[]{}:(),;[]', Punctuation),
+            (r'\\\n', Text),
+            (r'\\', Text),
+            (words(_operator, suffix=r'\b'), Operator.Word),
+            (r'<>|==|->|<=|>=|\?|[-~+/*%=<>&^|.!]', Operator),
+            include('keywords'),
+            include('builtins'),
+            include('name'),
+            (r'("(\\\\|\\"|[^"])*")|(\'(\\\\|\\\'|[^\'])*\')', String),
```

Comment:
This looks like you're trying to match strings and allow escapes inside. This is error prone (we've had a bunch of issues w.r.t. to catastrophic backtracking and strings) to put into a single regex. Can you please use state-based processing here? I.e. push/pop when entering/leaving a string?